text/2020-04-23-rawkv-txnkv-mode-switching.md

```diff
@@ -0,0 +1,63 @@
+# Title
+
+## Summary
+
+This RFC proposes to forbid users using RawKV API on the TiKV cluster by default,
+and provide a way for users to enable it.
+
+## Motivation
+
+Some painful incidents happened before and still happening,
+because of the mixed-using of RawKV interface and TransactionKV interface,
+some key-values were overwritten by RawKV,
+which originally written by TransactionKV, then led to data corruption.
+
+We should add more warnings about this in our documents,
+but it's not enough, documents could easily be ignored.
+
+Disable RawKV by default could prevent people from misuse without noticing,
+when users need to step into the danger zone,
+they need to explicitly open the guarded switch by command.
+And they could shut down the switch anytime to gain more safety.
+
+## Detailed design
+
+Add a `kv-mode` flag to the cluster, this flag should be persisted,
+each member of the TiKV cluster should see the same value at the same time,
+so it could be:
+
+- Store in a special key in the underlying engine and use a new command
+  to manipulate it
+- Use the dynamic configuration feature to achieve our goal.
+
+Since the dynamic configuration is already supported, it should be good for this.
+
+`kv-mode` should be an enum:
+
+- `ModeTxnKV` indicates that the RawKV commands could not be called.
```

Comment:
Do we still need to allow RawKV access? I guess the RawKV interface was added for debugging purpose at first..